web/src/fetch_status.ts

```diff
@@ -141,4 +141,10 @@ export class FetchStatus {
             max_id_for_messages(messages),
         );
     }
+
+    copy_status(fetch_status: FetchStatus): void {
+        this._found_newest = fetch_status.has_found_newest();
+        this._found_oldest = fetch_status.has_found_oldest();
+        this._expected_max_message_id = fetch_status._expected_max_message_id;
```

Comment:
Should `_history_limited` be included? I think you need a comment on the `loading` ones as to why they aren't set here.

Reply:
Yes, somehow I missed that.